pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -138,3 +158,84 @@ func (r *HelmWatcherReconciler) reconcileDelete(ctx context.Context, repository
 	// Stop reconciliation as the object is being deleted
 	return ctrl.Result{}, nil
 }
+
+// sendEvent emits a Kubernetes event and forwards the event to notification controller if configured.
+func (r *HelmWatcherReconciler) sendEvent(ctx context.Context, hr *sourcev1.HelmRepository, revision, severity, msg string) {
+	log := logr.FromContextOrDiscard(ctx)
```

Comment:
are we not already doing this in the reconciliation loop? can we not just pass the `log` in? Its also nice that the `log` in the reconciliation loop has the helm repo details attached